apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -324,18 +324,22 @@ export class PublishAction extends BaseRushAction {
             return;
           }
 
-          // Do not tag packages that already exist. This will fail with a fatal error.
-          if (this._packageExists(packageConfig)) {
+          const packageVersion: string = packageConfig.packageJson.version;
+
+          // Do not create a new tag if one already exists, this will result in a fatal error
+          if (git.hasTag(packageConfig)) {
+            console.log(`Not tagging ${packageName}@${packageVersion}. A tag already exists for this version.`);
             return;
           }
 
-          git.addTag(!!this._publish.value && !this._registryUrl.value, packageName, packageConfig.packageJson.version);
+          git.addTag(!!this._publish.value, packageName, packageVersion);
           updated = true;
         };
 
         if (this._pack.value) {
           // packs to tarball instead of publishing to NPM repository
           this._npmPack(packageName, packageConfig);
+          // Do not tag packages that already exist. This will fail with a fatal error.
```

Comment:
Are we missing a check here?

Reply:
No, the code moved but the comment didn't. d'oh. Removing the comment.